### DISCUSSION TITLE
Is it possible to contrast tab-item.label-content and tab-item.counter font colors against the background color set by the TST Colored Tabs extension

### DISCUSSION THREAD

**Discussion by bqback** [piroor/treestyletab#3632] [Q&A] [answered]
Title is pretty much the entire question. I figured out how to change font color based on [this](https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#highlight-active-tab) snippet, but I think TST either doesn't support `color-contrast()` (wouldn't be surprising based on its status on Mozilla MDN) or I'm not grabbing the background color right (wouldn't be surprising based on my limited CSS knowledge).
I also couldn't get mix-blend-mode to work either.

It's likely worth noting I have
```css
tab-item.active .background {
  opacity: 0;
}
```
in the TST stylesheet because I couldn't figure out how to remove the white background on active tabs. This probably shouldn't impact regular `tab-item` rules, but I couldn't get those to work.

**Comment by irvinm** [comment]
Firefox (not a TST limitation) does not support color-contrast() -> https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-contrast

I haven't looked into it, but could contrast() help?  https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/contrast

**Reply by bqback** [reply]
> Firefox (not a TST limitation) does not support color-contrast()

My bad, I think I misread Safari supporting it experimentally as Firefox supporting it experimentally. Oh well, maybe in the not so distant future...

`contrast()` seems to be applying to an element as a filter, and if I had dark text against a dark background, maxing out contrast would only marginally improve readability. Guess I'll just have to decide between dark backgrounds with bright text and light backgrounds with dark text (maybe former on inactive and latter on active?) and stick to it until `color-contrast()` is supported.

This more or less answers my question, so if no one else comes by with some hacky idea, I'll mark this as an answer, thank you.

**Reply by irvinm** [reply, marked as answer]
So I looked at how TST Colored Tabs is working and they are basically just generating CSS for the background to be passed over to TST a little later.

https://github.com/MurzNN/TST-Colored-tabs/blob/b6ebf80d2603e15a89f5b9816057e7b19cf73fbb/background.js#L111C1-L116C6

```
    if(ColoredTabs.state.hostsMatchColor !== undefined) {
      ColoredTabs.state.hostsMatchColor.forEach((element, index) => css += `tab-item.coloredTabsHostMatch` + index + ` tab-item-substance {background-color: ` + element + `;}`);
    }
    if(ColoredTabs.state.hostsRegexpColor !== undefined) {
      ColoredTabs.state.hostsRegexpColor.forEach((element, index) => css += `tab-item.coloredTabsHostRegexp` + index + ` tab-item-substance {background-color: ` + element + `;}`);
    }
```
I don't think there is a way from TST CSS to "read the color value of the background" and set the "text color" based on that so it had enough contrast.  HOWEVER, I think it wouldn't be too hard within the addon (Request feature enhancement) to add an option to also set the text color as part of the same call using some type of filter BASED on the background color being set.

You could also just ask TST Colored Tabs to add an option to just set the background AND the text to the same color and then you could just add TST CSS to apply a filter to the text to achieve their desired variation of it.  [Ex: text ... inverse/contrast(0.75)]
